Lib/test/test_urlparse.py

```diff
@@ -1402,16 +1402,34 @@ def test_issue14072(self):
         self.assertEqual(p2.path, '+31641044153')
 
     def test_invalid_bracketed_hosts(self):
```

Comment:
This refactoring, personally makes it hard to review. Could you please share the description, which existing test cases behavior are changing now (if any)? If you don't get to it, I will try to find out too. (But this can be considered for future refactors of test code).

Reply:
Sure - here are the differences in the tests:
- urlsplit and urlparse are now tested (not just urlsplit)
- strings and bytes are now tested (not just strings)
- four new test cases are added (the first ten are the same as they were before)

If I need to split the refactoring into a separate PR, I can do that.